app/src/main/kotlin/com/simplemobiletools/flashlight/activities/BrightDisplayActivity.kt

```diff
@@ -27,40 +45,65 @@ class BrightDisplayActivity : SimpleActivity() {
                 WindowManager.LayoutParams.FLAG_FULLSCREEN
         )
 
-        useDynamicTheme = false
         super.onCreate(savedInstanceState)
-        setContentView(binding.root)
-        supportActionBar?.hide()
-        setBackgroundColor(config.brightDisplayColor)
-
-        binding.brightDisplayChangeColor.setOnClickListener {
-            ColorPickerDialog(this, config.brightDisplayColor, true, currentColorCallback = {
-                setBackgroundColor(it)
-            }) { wasPositivePressed, color ->
-                if (wasPositivePressed) {
-                    config.brightDisplayColor = color
-
-                    val contrastColor = color.getContrastColor()
-                    binding.brightDisplayChangeColor.apply {
-                        setTextColor(contrastColor)
-                        background.applyColorFilter(contrastColor)
-                    }
-                } else {
-                    setBackgroundColor(config.brightDisplayColor)
+        enableEdgeToEdgeSimple()
+        setContent {
+            AppThemeSurface {
+                val colorPickerDialogState = rememberAlertDialogState().apply {
+                    ColorPicker()
                 }
+
+                ScreenContent(colorPickerDialogState)
             }
         }
+    }
+
+    @Composable
+    private fun AlertDialogState.ColorPicker() {
+        val brightDisplayColor by preferences.brightDisplayColorFlow.collectAsStateWithLifecycle(Color.WHITE)
```

Comment:
```kotlin
        val brightDisplayColor by preferences.brightDisplayColorFlow.collectAsStateWithLifecycle(preferences.brightDisplayColor)
```

Reply:
we should not have hardcoded values for default preferences when a flow is created, they should be read from the default prefs